bolt-cli/src/common/mod.rs

```diff
@@ -131,12 +133,41 @@ pub fn request_confirmation() {
                 std::process::exit(0);
             }
             InquireError::OperationInterrupted => {
-	       // Triggered a SIGINT via Ctrl-C
-	       std::process::exit(130);     
+                // Triggered a SIGINT via Ctrl-C
+                std::process::exit(130);
             }
             _ => {
                 error!("aborting due to unexpected error: {}", err);
                 std::process::exit(1);
             }
         })
 }
+
+/// Determines the RPC URL to use. If `dry_run` is enabled, it spawns an `Anvil` instance and
+/// returns its endpoint. Otherwise, returns the original `rpc_url`.
+pub(crate) fn handle_rpc_dry_run(
+    rpc_url: Url,
+    dry_run: bool,
+) -> eyre::Result<(Url, Option<AnvilInstance>)> {
+    if !dry_run {
+        return Ok((rpc_url, None));
+    }
+
+    let anvil = Anvil::new()
+        .fork(rpc_url)
+        .try_spawn()
+        .map_err(|e| eyre::eyre!("[dry-run] Failed to spawn Anvil: {}", e))?;
+
+    let anvil_url = anvil.endpoint_url();
+    info!("[dry-run] Anvil endpoint URL: {}", anvil_url);
+
+    Ok((anvil_url, Some(anvil)))
+}
+
+/// drop provided `AnvilInstance` to control resource consumption
+pub fn shutdown_anvil(anvil: Option<AnvilInstance>) {
+    if let Some(anvil_instance) = anvil {
+        info!("[dry-run] Shutting down Anvil instance.");
+        drop(anvil_instance);
+    }
+}
```

Comment:
Let me know if I got this right: you need this function (or more generally to manually drop anvil) and call it at the end of a command otherwise the compiler insert the drop too early, right after `handle_rpc_dry_run` is called because it is no more used.